custom_components/ha_washdata/profile_store.py

```
@@ -5623,17 +6030,323 @@ async def async_import_data(self, payload: dict[str, Any]) -> dict[str, Any]:
+        selected_past = _apply_id_filter(src_past, real_id_filter)
+
+        # Replace-mode anti-data-loss guard: never wipe a destination unless the file has a
+        # non-empty SELECTED set to refill it. Guarding on the post-filter sets (not raw
+        # src_*) means a stale/malformed id selection that matches nothing aborts here instead
+        # of emptying the local list. Profiles are never wiped (per-name overwrite/copy), so
+        # a profiles-only replace needs no guard.
+        if mode == "replace":
+            if "reference_cycles" in cats and not selected_refs:
+                raise ValueError(
+                    "Import payload contains no reference cycles — aborting to prevent data loss"
+                )
```

Ruff (0.15.21)

[warning] Avoid specifying long messages outside the exception class

(TRY003)